### PR TITLE
Fix reverse preference in matter window covering

### PR DESCRIPTION
**Pull request by nickolas-deboom**
# Type of Change

- [ ] WWST Certification Request
     - If this is your first time contributing code:
          - [ ] I have reviewed the README.md file
          - [ ] I have reviewed the CODE_OF_CONDUCT.md file
          - [ ] I have signed the CLA
     - [ ] I plan on entering a WWST Certification Request or have entered a request through the WWST Certification console at developer.smartthings.com
- [x] Bug fix
- [ ] New feature
- [ ] Refactor

# Checklist

- [x] I have performed a self-review of my code
- [x] I have commented my code in hard-to-understand areas
- [x] I have verified my changes by testing with a device or have communicated a plan for testing
- [ ] I am adding new behavior, such as adding a sub-driver, and have added and run new unit tests to cover the new behavior

# Description of Change

This change inverts the interpretation of open/closed when the Reverse preference is set. This means that when the preference is set, a `windowShadeLevel` of 100% is interpreted as Closed and 0% as Open. This approach replaces a previous change that inverted the percentage values received from the device, which was more complex and has some edge cases that caused occasional issues.

# Summary of Completed Tests

Tested with VDA window covering, Zemismart MotionBlinds, and updated unit tests.

### CODE REVIEW

**Review by HunsupJung — APPROVED** [review]

File: drivers/SmartThings/matter-window-covering/src/matter-window-covering-position-updates-while-moving/init.lua [hunk 1]
```diff
@@ -136,6 +130,16 @@ local function current_status_handler(driver, device, ib, response)
   end
 end
 
+local function level_attr_handler(driver, device, ib, response)
+  -- VDA reports CurrentLevel but always with a value of 0, so ignore these reports
+  if (device.manufacturer_info.vendor_id == 0x10e1 and device.manufacturer_info.product_id == 0x1005) or
```
Comment (z-michel) [inline drivers/SmartThings/matter-window-covering/src/matter-window-covering-position-updates-while-moving/init.lua]: Discussed offline but let's remove this and ask VDA to fix if there is problematic behavior.

Comment (nickolas-deboom) [reply drivers/SmartThings/matter-window-covering/src/matter-window-covering-position-updates-while-moving/init.lua]: Removed in latest commit

File: drivers/SmartThings/matter-window-covering/src/init.lua [hunk 2]
```diff
@@ -169,7 +168,7 @@ end
 -- move to shade level between 0-100
 local function handle_shade_level(driver, device, cmd)
   local endpoint_id = device:component_to_endpoint(cmd.component)
-  local lift_percentage_value = reverse_polarity_if_needed(device, cmd.args.shadeLevel)
```
Comment (hcarter-775) [inline drivers/SmartThings/matter-window-covering/src/init.lua:172]: I see this removes any kind of reverse functionality from the shade level handler. What is the context for removing this?

File: drivers/SmartThings/matter-window-covering/src/init.lua [hunk 3]
```diff
@@ -180,7 +179,7 @@ end
 -- move to shade tilt level between 0-100
 local function handle_shade_tilt_level(driver, device, cmd)
   local endpoint_id = device:component_to_endpoint(cmd.component)
-  local tilt_percentage_value = reverse_polarity_if_needed(device, cmd.args.level)
```
Comment (hcarter-775) [inline drivers/SmartThings/matter-window-covering/src/init.lua:183]: Same with the tilt level as my above question about the shade level

File: drivers/SmartThings/matter-window-covering/src/init.lua (hunk 2 above)
Comment (nickolas-deboom) [reply drivers/SmartThings/matter-window-covering/src/init.lua:172]: Rather than inverting the percentage values reported by the device, the simpler approach (and more in line with the initial request) is to interpret 0 as open and 100 as closed if the Reverse preference is enabled (and also to interpret the Open capability command as Close and vice-versa).

**Comment by nickolas-deboom** [comment]
> Im also curious, have we been able to test this on a non-vda device, since the behavior of the vda through the subdriver is different to that of a regular device? What is the thinking there?

Good call out, I have a physical blind I can test with

**Comment by nickolas-deboom** [comment]
> > Im also curious, have we been able to test this on a non-vda device, since the behavior of the vda through the subdriver is different to that of a regular device? What is the thinking there?
> 
> Good call out, I have a physical blind I can test with

Actually, this has already been successfully tested with a Zemismart MotionBlind

**Review by hcarter-775 — APPROVED** [review]
